virtwho/config.py

```diff
@@ -685,29 +694,487 @@ def addConfig(self, config):
         self._configs.append(config)
 
 
-def getOptions(section, parser):
-    options = {}
-    for option in parser.options(section):
-        options[option] = parser.get(section, option)
-    return options
-
-
-def getSections(parser):
-    sections = {}
-    for section in parser.sections():
-        try:
-            sections[section] = getOptions(section, parser)
-        except NoOptionError:
-            sections[section] = {}
-    return sections
-
-
-def parseFile(filename, logger=None):
+def parseFile(filename):
     # Parse a file into a dict of section_name: options_dict
     # options_dict is a dict of option_name: value
     parser = StripQuotesConfigParser()
     fname = parser.read(filename)
     if len(fname) == 0 and logger:
         logger.error("Unable to read configuration file %s", filename)
-    sections = getSections(parser)
+    sections = parser._sections
+    for section in sections:
+        if '__name__' in sections[section]:
+            del sections[section]['__name__']
     return sections
+
+# Helper methods used to validate parameters given to virt-who
+def str_to_bool(value):
+    if isinstance(value, bool):
+        return value
+    if isinstance(value, str):
+        return value.strip().lower() in ['yes', 'true', 'on', '1']
+    raise ValueError("Unable to convert value to boolean")
+
+
+def non_empty_string(value):
+    if not isinstance(value, str):
+        raise TypeError("Value is not a string")
+    if not value:
+        raise ValueError("String is empty")
+    return value
+
+
+def readable(path):
+    if not os.access(path, os.R_OK):
+        raise ValueError("Path '%s' is not accessible" % path)
+    return path
+
+
+def accessible_file(path):
+    readable(path)
+    if not os.path.isfile(path):
+        raise ValueError("Path '%s' does not appear to be a file" % path)
+    return path
+
+
+def accessible_dir(path):
+    readable(path)
+    if not os.path.isdir(path):
+        raise ValueError("Path '%s' does not appear to be a directory" % path)
+
+
+def empty_or_accessible_files(paths):
```

Comment:
I would add some doc string for this method, because the name is no self-explanatory.